src/sys/wait.rs

```diff
@@ -161,34 +161,38 @@ fn continued(status: i32) -> bool {
     unsafe { libc::WIFCONTINUED(status) }
 }
 
-fn decode(pid: Pid, status: i32) -> WaitStatus {
-    if exited(status) {
-        WaitStatus::Exited(pid, exit_status(status))
-    } else if signaled(status) {
-        WaitStatus::Signaled(pid, term_signal(status), dumped_core(status))
-    } else if stopped(status) {
-        cfg_if! {
-            if #[cfg(any(target_os = "linux", target_os = "android"))] {
-                fn decode_stopped(pid: Pid, status: i32) -> WaitStatus {
-                    let status_additional = stop_additional(status);
-                    if syscall_stop(status) {
-                        WaitStatus::PtraceSyscall(pid)
-                    } else if status_additional == 0 {
-                        WaitStatus::Stopped(pid, stop_signal(status))
-                    } else {
-                        WaitStatus::PtraceEvent(pid, stop_signal(status), stop_additional(status))
+impl WaitStatus {
+    /// Convert a raw `wstatus` as returned by `waitpid`/`wait` into a `WaitStatus`.
+    /// Returns an `Error` corresponding to `EINVAL` for invalid status values.
```

Comment:
This needs a blank line between these 2 comments. That makes the first line the short summary line in the generated docs. Additionally since the second talks about errors, there's a standard way to display this into under a `# Errors` header. The rust doccomment documentation talks about those sections.